compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -434,8 +437,12 @@ static void compileInternal(ModuleContext moduleContext, CompilerContext compile
             packageCache.putSymbol(pkgNode.packageID, pkgNode.symbol);
             compilerPhaseRunner.performTypeCheckPhases(pkgNode);
         } catch (Throwable t) {
-            assert false : "Compilation failed due to" +
-                    (t.getMessage() != null ? ": " + t.getMessage() : " an unhandled exception");
+            assert false : "Compilation failed due to " + ((Supplier<String>)() -> {
+                StringWriter errors = new StringWriter();
+                t.printStackTrace(new PrintWriter(errors));
+                return errors.toString();
+            }).get();
+
```

Comment:
Is it fine to add the stacktrace here? Otherwise, I couldn't find out why the compilation failed exactly. I also think this assert is only triggered while testing and is disabled for the normal distribution.

Reply:
Yes assertions only trigger when you pass the flag to jvm (which we normally only do as a part tests). Personally I put a conditional breakpoint at the assert and run the debugger, but I don't think there is any harm in printing the stack trace.